src/Documentation/SidebarMenu/SidebarMenu.js

```diff
@@ -35,6 +38,9 @@ function calculateHeight({ activePaths, path }) {
   return height
 }
 
+/**
+ * <SidebarMenuItem> (Pure) Pomponent for <SidebarMenu>
+ */
 class SidebarMenuItem extends React.PureComponent {
```

Comment:
Why is this a `PureComponent`? I'm not sure this component satisfies the guidelines in [this SO answer](https://stackoverflow.com/a/43936258/761963) (assuming it's correct).

Reply:
@jorgeorpinel which of the guidelines does it violate? Its state anyway doesn't change.

Reply:
Please correct me if I'm wrong, still learning React, but I don't see any of them here:

✖️ State/Props should be an immutable object
✖️ State/Props should not have a hierarchy (not sure what this means actually)
✖️ You should call `forceUpdate` when data changes

But anyway, I'm not suggesting to change this. Mainly I want to know why this component is a `PureComponent`. Do you know @algomaster99?

p.s. There's another 2 `PureComponent` subclasses in the docs we could check.

Reply:
@jorgeorpinel It's not exactly works like this. Basic idea is that every time parent component is rerendered, it will cause all it's child components to rerender too. Sometimes this rerender process is cheap, but in some other cases it can be quire resource consuming (for example if we call markdown component on each rerender it will try to parse/stringify document with remark every time, which is not fast).

In this case we have some listeners for the scroll in sidebar and there is some changes that fired on page scroll in `docs.js`, because of that most docs page children components are optimized to prevent unnecessary rerenders.

I have an article about the React render process details, if you are interested: https://evilmartians.com/chronicles/optimizing-react-virtual-dom-explained

Reply:
Super informatie. Thanks @iAdramelk !